pkg/pullsync/pullsync.go

```diff
@@ -246,24 +247,22 @@ func (s *Syncer) SyncInterval(ctx context.Context, peer swarm.Address, bin uint8
 
 	if len(chunksToPut) > 0 {
 
-		if to != MaxCursor { // historical syncing
-			s.rate.Add(len(chunksToPut))
-		}
-
 		s.metrics.DbOps.Inc()
 
-		if err := s.storage.Put(ctx, storage.ModePutSync, chunksToPut...); err != nil {
-			return topmost, fmt.Errorf("delivery put: %w", err)
+		putter := s.store.ReservePutter(ctx)
+		for _, c := range chunksToPut {
+			if err := putter.Put(ctx, c); err != nil {
+				_ = putter.Cleanup()
+				return topmost, 0, fmt.Errorf("delivery put: %w", err)
```

Comment:
you can do the cleanup in-line here `errors.Join(err, putter.Cleanup())` also capturing the error if any on Cleanup